src/render/SDL_palette_sw_c.h

```diff
@@ -0,0 +1,55 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2022 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+#ifndef SDL_palette_sw_c_h_
+#define SDL_palette_sw_c_h_
+
+#include "../SDL_internal.h"
+
+#include "SDL_video.h"
+
+/* This is the software implementation of the paletted texture support */
+
+struct SDL_SW_PaletteTexture
+{
+    SDL_Surface *surface;
+    SDL_Palette *palette;
+
+    SDL_Surface *display;
+};
+
+typedef struct SDL_SW_PaletteTexture SDL_SW_PaletteTexture;
+
+SDL_SW_PaletteTexture *SDL_SW_CreatePaletteTexture(Uint32 format, int w, int h);
```

Comment:
All these functions miss Doxygen comments and also please add empty lines between them.

Reply:
Doxygen comments don't generally seem to be used in internal headers such as this one.